### PR TITLE
Micro Reddit

### PR DESCRIPTION
**Pull request by OlawaleJoseph**
This a project based on understanding the concepts of ActiveRecord in Ruby on Rails.

### Testing
- Get a local copy of this repo
  Open a terminal and run: ```git clone git@github.com:OlawaleJoseph/micro-reddit.git```
- Open a terminal and run ```bunle install```
- Run ```yarn install --check-files```
- Run ```rails c ```
- Create the following Entities to test
  - User (name: string)
  - Post (body: string)
  - Comment (body: string)

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+
+  validates :body, presence: true, length: { maximum: 1000 }
+  validates :user_id, presence: true
```
Comment (uryelah) [inline app/models/post.rb]: - [x] The belongs_to relation already validates the user_id for you, so this line is unnecessary.

- [x] You are missing the title validation, the post should have both a body and a title.

**Review by uryelah — CHANGES_REQUESTED** [review]
Hi **Wale Adedeko & Mark Kiptubei**, I'm Sarah your reviewer for this PR 👩‍🚀 

Great job with the relations and validations! Everything but one thing seems to be fine!
Check the comments below for the required changes and keep on excelling! 

# Required Changes 🏗️ 🐛 💡

- [ ] Your post is missing a title column, make sure to add the necessary migration and validation.

- [ ] Make sure to capitalize your commit for [best practices](https://chris.beams.io/posts/git-commit/).

-----

 **Kindly fix the pointed out issues and comments below before requesting the next code review**

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

File: README.md [hunk 2]
```diff
@@ -1,24 +1,39 @@
...
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+# Micro Reddit
 
-Things you may want to cover:
+## Purpose
 
-* Ruby version
+> This is a project done with the aim of understanding how ActiveRecord works in rails.
```
Comment (mosesogwo) [inline README.md:5]: You could add a description of the models and associations present in the project.

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hello! Great job on this :+1:. Just a few changes and you can :rocket:**

:point_right: Add validations to prevent creating a comment with an invalid body, say empty strings.

:point_right: See comment on your README

_**Kindly make these changes and submit another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

**Review by stratospherique — APPROVED** [review]
### Great work implementing the suggested changes :clap: Congrats, the project is approved :heavy_check_mark: You may merge :+1: :+1: 

Keep up the good work :muscle: 
**_Ahmed Mahfoudh_**

**Pull request merged**